cmd/otel-allocator/config/config.go

```diff
@@ -129,3 +129,13 @@ func ParseCLI() (CLIConfig, error) {
 	cLIConf.ClusterConfig = clusterConfig
 	return cLIConf, nil
 }
+
+// ValidateConfig validates the cli and file configs together.
+func ValidateConfig(config *Config, cliConfig *CLIConfig) error {
+	if !*cliConfig.PromCRWatcherConf.Enabled &&
+		(config.Config == nil ||
+			len(config.Config.ScrapeConfigs) == 0) {
+		return fmt.Errorf("either at least one scrape config must be defined, or Prometheus CR watching must be enabled")
```

Comment:
Small nit
```suggestion
		return fmt.Errorf("at least one scrape config must be defined, or Prometheus CR watching must be enabled")
```

Reply:
i'd prefer us to actually split up this if in to three statements. Although it's more LOC we'll be able to give a more descriptive error back which I think would be better.

Reply:
I wrote it this way anticipating that we might want to add more checks in the future, in which case they all need to be negative checks (as in "this error condition doesn't occur"). And I can't really rewrite it into three separate statements, as it's logically an OR.

I could make it nested instead, but I'm not sure if you can really make the error message any more useful that way.

Reply:
I rewrote this condition to be clearer, let me know if this helps.